src/imitation/util/networks.py

```diff
@@ -0,0 +1,150 @@
+"""Helper methods to build and run neural networks."""
+
+import collections
+import contextlib
+from typing import Callable, Dict, Iterable, Optional, Sequence, Tuple
+
+import gym
+import tensorflow as tf
+from stable_baselines.common.input import observation_input
+
+# TODO(adam): we should use typing.OrderedDict introduced in Python 3.7.2
+# As of 2020-05-07, pytype does not understand OrderedDict; worth retrying.
+LayersDict = Dict[str, tf.layers.Layer]
+
+
+def build_mlp(
+    hid_sizes: Iterable[int],
+    name: Optional[str] = None,
+    activation: Optional[Callable] = tf.nn.relu,
+    initializer: Optional[Callable] = None,
+) -> LayersDict:
+    """Constructs an MLP, returning an ordered dict of layers."""
+    layers = collections.OrderedDict()
+
+    # Hidden layers
+    for i, size in enumerate(hid_sizes):
+        key = f"{name}_dense{i}"
+        layer = tf.layers.Dense(
+            size, activation=activation, kernel_initializer=initializer, name=key
+        )  # type: tf.layers.Layer
+        layers[key] = layer
+
+    # Final layer
+    layer = tf.layers.Dense(
+        1, kernel_initializer=initializer, name=f"{name}_dense_final"
+    )  # type: tf.layers.Layer
+    layers[f"{name}_dense_final"] = layer
+
+    return layers
+
+
+def sequential(inputs: tf.Tensor, layers: LayersDict,) -> tf.Tensor:
+    """Applies a sequence of layers to an input."""
```

Comment:
Until we have OrderedDict typing, could you an `assert isinstance(layers, OrderedDict)` here?

Reply:
On reflection I think I should just drop the warning comment and ignore this. We're assuming Python >3.7 in `setup.py` and dictionaries are guaranteed to be insertion ordered from 3.7 and above: https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6

Reply:
Oh wow, OrderedDict becomes much less important then.

https://stackoverflow.com/questions/50872498/will-ordereddict-become-redundant-in-python-3-7